### PR TITLE
Modified & add models and requests to support CloudStack network API

### PR DESCRIPTION
**Pull request by resouer**
Modified & add models and requests to some missing api which is used by Cloud Foundry in bosh-cloudstack-cpi project.

Influenced files and added tests:
```
 create mode 100644 lib/fog/cloudstack/models/compute/firewall.rb
 create mode 100644 lib/fog/cloudstack/models/compute/firewalls.rb
 create mode 100644 lib/fog/cloudstack/models/compute/ipaddress.rb
 create mode 100644 lib/fog/cloudstack/models/compute/ipaddresses.rb
 create mode 100644 lib/fog/cloudstack/models/compute/key_pair.rb
 create mode 100644 lib/fog/cloudstack/models/compute/key_pairs.rb
 create mode 100644 lib/fog/cloudstack/models/compute/nat.rb
 create mode 100644 lib/fog/cloudstack/models/compute/nats.rb
 create mode 100644 lib/fog/cloudstack/models/compute/network.rb
 create mode 100644 lib/fog/cloudstack/models/compute/networks.rb
 create mode 100644 lib/fog/cloudstack/models/compute/ostype.rb
 create mode 100644 lib/fog/cloudstack/models/compute/ostypes.rb
 create mode 100644 lib/fog/cloudstack/models/compute/vlan.rb
 create mode 100644 lib/fog/cloudstack/models/compute/vlans.rb
 create mode 100644 tests/cloudstack/requests/assosiate_ip_address_tests.rb
 create mode 100644 tests/cloudstack/requests/create_vlan_range_tests.rb
 create mode 100644 tests/cloudstack/requests/delete_vlan_range_tests.rb
 create mode 100644 tests/cloudstack/requests/disable_static_nat_tests.rb
 create mode 100644 tests/cloudstack/requests/disassosiate_ip_address_tests.rb
 create mode 100644 tests/cloudstack/requests/enable_static_nat_tests.rb
 create mode 100644 tests/cloudstack/requests/list_vlan_range_tests.rb
```

### CODE REVIEW

**Comment by resouer** [comment]
@plribeiro3000 thx for your comments! Is everything OK now?

**Comment by plribeiro3000** [comment]
@resouer Sorry for the delay. Taking a look now.

File: lib/fog/cloudstack/models/compute/firewall.rb [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+                    options = {
+                            'ipaddressid'        => ip_address_id,
+                            'protocol'           => protocol,
+                            'cidrlist'           => cidr_list,
+                        }
+                    options.merge!('startport' => start_port) if start_port
+                    options.merge!('endport' => end_port) if end_port
+                    options.merge!('icmptype' => icmp_type) if icmp_type
+                    options.merge!('icmpcode' => icmp_code) if icmp_code
```
Comment (plribeiro3000) [inline lib/fog/cloudstack/models/compute/firewall.rb]: Perhaps what you want to do here is the same you can accomplish with [`all_associations_and_attributes`](https://github.com/fog/fog-core/blob/master/lib/fog/core/attributes.rb#L94). 

The `nil` options need to be discarded? You can see how i used it [here](https://github.com/fog/fog-xenserver/blob/master/lib/fog/compute/xen_server/models/instance_methods.rb#L24).

File: lib/fog/cloudstack/models/compute/image.rb [hunk 2]
```diff
@@ -66,6 +66,28 @@ def destroy
...
+          options = {
+                  'displaytext' => display_text,
+                  'format' => format,
+                  'hypervisor' => hypervisor,
+                  'name' => name,
+                  'ostypeid' => os_type_id,
+                  'url' => url,
+                  'zoneid' => zone_id,
+                  'isfeatured' => is_featured
+          }
```
Comment (plribeiro3000) [inline lib/fog/cloudstack/models/compute/image.rb]: Perhaps you can use `all_associations_and_attributes` here as well and D.R.Y it up.

**Comment by plribeiro3000** [comment]
@resouer The code seems to be ok now. Just some comments related to a new api added to `fog-core` to make it easier to maintain the codebase. Can you give it a try?

LMK if you need any directions related to it.

**Comment by plribeiro3000** [comment]
@resouer Ping me back when you commit here so i can take a look again.

**Comment by plribeiro3000** [comment]
Closing due inactivity.

Most part of the code seems to be ok but it lacks some style standards like [this](https://github.com/fog/fog/pull/3362/files#diff-1ade1ed8ec57d5c9ccea45753f0d9697R16) and some improvements using the new features available in `fog-core`.

Ping back here when you have time to finish this. Thanks!

**Comment by geemus** [comment]
Thanks!

**Pull request closed without merging**